src/contracts/core/DelegationManager.sol

```diff
@@ -254,11 +252,13 @@ contract DelegationManager is Initializable, OwnableUpgradeable, Pausable, Deleg
      * @param staker The address to decrease the delegated shares for their operator.
      * @param strategies An array of strategies to crease the delegated shares.
      * @param shares An array of the number of shares to decrease for a operator and strategy.
+     * @param undelegateIfPossible If marked 'true', then this contract will check if the `staker` can undelegate, and undelegate them if possible.
+     * If the check fails, then the `staker` will simply remain delegated.
      * 
      * @dev *If the staker is actively delegated*, then decreases the `staker`'s delegated shares in each entry of `strategies` by its respective `shares[i]`. Otherwise does nothing.
-     * @dev Callable only by the StrategyManager.
+     * @dev Callable only by the StrategyManager or EigenPodManager.
      */
-    function decreaseDelegatedShares(address staker, IStrategy[] calldata strategies, uint256[] calldata shares)
+    function decreaseDelegatedShares(address staker, IStrategy[] calldata strategies, uint256[] calldata shares, bool undelegateIfPossible)
```

Comment:
Is there a reason we're not auto-undelegating a user if their shares are being decreased by an amount that zeros out their entire share balance in all the strategies instead of undelegateIfPossible?

Reply:
> Adding undelegateIfPossible feels like it adds a bit of complexity
yes, sorry for lack of clarity, this PR to this branch further simplifies / deletes this https://github.com/Layr-Labs/eigenlayer-contracts/pull/183